src/core/public/application/ui/app_container.tsx

```diff
@@ -65,28 +70,42 @@ export const AppContainer: FunctionComponent<Props> = ({
     }
     setAppNotFound(false);
 
+    setIsMounting(true);
     if (mounter.unmountBeforeMounting) {
       unmount();
     }
 
     const mount = async () => {
-      unmountRef.current =
-        (await mounter.mount({
-          appBasePath: mounter.appBasePath,
-          history: createScopedHistory(appPath),
-          element: elementRef.current!,
-          onAppLeave: handler => setAppLeaveHandler(appId, handler),
-        })) || null;
+      setShowSpinner(true);
+      try {
+        unmountRef.current =
+          (await mounter.mount({
+            appBasePath: mounter.appBasePath,
+            history: createScopedHistory(appPath),
+            element: elementRef.current!,
+            onAppLeave: handler => setAppLeaveHandler(appId, handler),
+          })) || null;
+      } catch (e) {
+        // TODO: add error UI
```

Comment:
Will do as a follow up, most likely using the same basic UI as the "app not found" UI

Reply:
I don't want to use the fatal error service here since it's possible that other applications are not broken. No need to crash the entire frontend just because one app cannot mount properly.